lib/chkhash.c

```diff
@@ -0,0 +1,70 @@
+#include <config.h>
+
+#include "chkhash.h"
+
+#include <regex.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+/*
+ * match_regex - return true if match, false if not
+ */
+bool 
+match_regex(const char *pattern, const char *string) 
+{
+	regex_t regex;
+	int result;
+
+	if (regcomp(&regex, pattern, REG_EXTENDED) != 0)
+		return false;
+
+	result = regexec(&regex, string, 0, NULL, 0);
+	regfree(&regex);
+
+	return result == 0;
+}
+
+/*
+ * is_valid_hash - check if the given string is a valid password hash
+ *
+ * Returns true if the string appears to be a valid hash, false otherwise.
+ */
+bool 
+is_valid_hash(const char *hash) 
+{
+	if (hash == NULL)
+		return false;
```

Comment:
This seems to be dead code.  `hash` is never NULL.  Please remove.  Otherwise, it can confuse programmers to think that a null pointer might be there sometimes.

Reply:
Here's why it can't be NULL:

<https://github.com/shadow-maint/shadow/blob/36debf3ccfa33ea9f3b0e562a133a22d1bdc87bd/src/chpasswd.c#L536-L544>